src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelMailsPage/ToolbarFilter/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button } from '@equinor/fusion-components';
+import { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import Personnel from '../../../../../../../models/Personnel';
+import styles from './styles.less';
+
+type ToolbarFilterProps = {
+    personnel: Personnel[];
+    setFilteredPersonnel: (filteredPersonnel: Personnel[]) => void;
+    filteredPersonnel: Personnel[];
+};
+type FilterKeys = 'noMail' | 'showMissingAD';
+type PersonnelFilter = (item: Personnel) => boolean;
+type Filters = {
+    filter: PersonnelFilter;
+    isSelected?: boolean;
+};
+type SelectedFilters = Record<FilterKeys, Filters>;
+
+const filterDefinitions: SelectedFilters = {
+    noMail: {
+        filter: (p) => !p.mail && !p.preferredContactMail,
+    },
+    showMissingAD: {
+        filter: (p) => p.azureAdStatus !== 'NoAccount',
+        isSelected: true,
+    },
+};
+const ToolbarFilter: FC<ToolbarFilterProps> = ({
+    personnel,
+    setFilteredPersonnel,
+    filteredPersonnel,
+}) => {
+    const [selectedFilters, setSelectedFilters] = useState<SelectedFilters>(filterDefinitions);
+
+    const personnelWithNoMail = useMemo(
+        () => filteredPersonnel.filter(filterDefinitions.noMail.filter).length,
+        [filteredPersonnel]
+    );
+
+    const personnelWithNoAd = useMemo(
+        () => personnel.filter((p) => !filterDefinitions.showMissingAD.filter(p)).length,
+        [personnel]
+    );
+
+    const filterPersonnel = useCallback(() => {
+        const activeFilters = Object.values(selectedFilters)
+            .filter((f) => f.isSelected)
```

Comment:
!!f?.isSelected instead, since the isSelected prop is optional?

Reply:
The filter will treat the undefined value as a false. The f, selectedFilter, cannot be undefined so the optional chaining should not be necessary

Reply:
Added !! for clearity